teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -3,27 +3,18 @@ package org.jetbrains.teamcity.rest
 import java.io.File
 import java.util.*
 
-abstract class TeamCityInstance {
-    abstract fun withLogResponses() : TeamCityInstance
-
-    abstract fun builds(): BuildLocator
-
-    abstract fun build(id: BuildId): Build
-    abstract fun build(buildType: BuildConfigurationId, number: String): Build
-    abstract fun buildConfiguration(id: BuildConfigurationId): BuildConfiguration
-    abstract fun vcsRoots(): VcsRootLocator
-    abstract fun vcsRoot(id: VcsRootId): VcsRoot
-    abstract fun project(id: ProjectId): Project
-    abstract fun rootProject(): Project
-
-    companion object {
-        @JvmStatic
-        fun guestAuth(serverUrl: String): TeamCityInstance = createGuestAuthInstance(serverUrl)
-
-        @JvmStatic
-        fun httpAuth(serverUrl: String, username: String, password: String): TeamCityInstance
-                = createHttpAuthInstance(serverUrl, username, password)
-    }
+interface TeamCityApi {
```

Comment:
This name looks confusing for me, actually every interface in this file is an API. Maybe we can keep the old name, and also keep the old class in API? We can implement factory methods via reflection to avoid compile-time dependency from api to impl, mark them as deprecated and provide a separate class in impl module with proper factory methods.